src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -80,27 +80,17 @@ internal unsafe IISHttpContext(PipeFactory pipeFactory, IntPtr pHttpContext)
 
                 var originalPath = RequestUriBuilder.DecodeAndUnescapePath(GetRawUrlInBytes());
 
-                // TODO: Read this from IIS config
-                // See https://github.com/aspnet/IISIntegration/issues/427
-                var prefix = "/";
                 if (KnownMethod == HttpApiTypes.HTTP_VERB.HttpVerbOPTIONS && string.Equals(RawTarget, "*", StringComparison.Ordinal))
                 {
                     PathBase = string.Empty;
                     Path = string.Empty;
                 }
-                // These paths are both unescaped already.
-                else if (originalPath.Length == prefix.Length - 1)
-                {
-                    // They matched exactly except for the trailing slash.
-                    PathBase = originalPath;
-                    Path = string.Empty;
-                }
                 else
                 {
-                    // url: /base/path, prefix: /base/, base: /base, path: /path
-                    // url: /, prefix: /, base: , path: /
-                    PathBase = originalPath.Substring(0, prefix.Length - 1);
-                    Path = originalPath.Substring(prefix.Length - 1);
+                    // Path and pathbase are unescaped by RequestUriBuilder
+                    // The UsePathBase middleware will modify the pathbase and path correctly
+                    PathBase = string.Empty;
+                    Path = originalPath;
```

Comment:
Is this correct? Does the path properly exclude the path base?

Reply:
We add the PathBase middleware right after the Server, just like IISIntegration did.  https://github.com/aspnet/IISIntegration/blob/dev/src/Microsoft.AspNetCore.Server.IISIntegration/IISSetupFilter.cs

Reply:
Maybe I misunderstand how it works but doesn’t that just set the path base? The path shouldn’t include the path base. What I mean by that is that if we go that route, we have to make sure that the way we’re getting the url from the underlying API excludes the path base.

If he path base is /foo and an http request comes in to /foo/bar. The path should be /bar and the path base should be /foo.

Is that the case?

Reply:
Yes that would be the case. The UsePathBase middleware will change the path and pathbase. https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Abstractions/Extensions/UsePathBaseMiddleware.cs#L58-L59